frontend/src/metabase/databases/utils/visibility.ts

```diff
@@ -0,0 +1,15 @@
+import { DatabaseData, Engine } from "metabase-types/api";
+
+export const getVisibleFields = (engine: Engine, values: DatabaseData) => {
+  const fields = engine["details-fields"] ?? [];
+
+  return fields.filter(field => {
+    const rules = field["visible-if"] ?? {};
+
+    return Object.entries(rules).every(([name, value]) =>
+      Array.isArray(value)
+        ? value.includes(values.details?.[name])
+        : value === values.details?.[name],
+    );
+  });
+};
```

Comment:
Responded in the comment. Seems not possible since the form is entirely backend driven and `visible-if` is used for different form fields of different (even custom) engines.